ci-operator/step-registry/ipi/deprovision/deprovision/ipi-deprovision-deprovision-commands.sh

```diff
@@ -4,11 +4,21 @@ set -o nounset
 set -o errexit
 set -o pipefail
 
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
 cluster_profile=/var/run/secrets/ci.openshift.io/cluster-profile
 export AWS_SHARED_CREDENTIALS_FILE=$cluster_profile/.awscred
 export AZURE_AUTH_LOCATION=$cluster_profile/osServicePrincipal.json
 
 echo "Deprovisioning cluster ..."
 cp -ar "${SHARED_DIR}" /tmp/installer
-openshift-install --dir /tmp/installer destroy cluster
-cp /tmp/installer/.openshift_install.log "${ARTIFACT_DIR}/"
+openshift-install --dir /tmp/installer destroy cluster &
+
+set +e
```

Comment:
So is the intent of this to continue even if it fails? If so, it's racy -- you can have the background fail before you `set +e`. Why are we doing this?

Reply:
I don't think `errexit`, even with `pipefail`, considers background processes.  I can't find a positive reference, but:

```
$ time bash -ec 'false & sleep 2; wait "$!"'

real    0m2.010s
user    0m0.004s
sys     0m0.007s
```

Reply:
> So is the intent of this to continue even if it fails?

Yes.  And capture the backgrounded command's exit status so we can return it later on.

> If so, it's racy -- you can have the background fail before you `set +e`.

Indeed.  As described in the PR topic comment, this approach is recycled from de3de20 (#6708).  But testing locally with `false &`, I can produce your race.  Will reroll to handle log collection via `trap` to avoid the race.

Reply:
Why not just

```sh
if ! openshift-install --dir /tmp/installer destroy cluster; then
    # failure 
else
    # not
fi
```

Reply:
we need to background it to get `trap` TERM handling; see 4472ace (#2680).

Reply:
Hmm.  I thought I'd reproduced the race in a shell earlier, but trying to nail this down for a commit message I'm having trouble:

```console
$ echo $BASH_VERSION
5.0.11(1)-release
$ cat test.sh
#!/bin/bash

set -o nounset
set -o errexit
set -o pipefail

"${@}" &
sleep 1
set +e
wait "$!"
ret="$?"
set -e
echo "post-wait gather"
exit "${ret}"
$ ./test.sh false
post-wait gather
$ echo $?
1
$ ./test.sh sleep 2
post-wait gather
$ echo $?
0
```

Can you provide an example that exposes the race?